PyInstaller/utils/osx.py

```diff
@@ -101,13 +101,30 @@ def get_macos_sdk_version(filename):
     header = binary.headers[0]
     # Find version command using helper
     version_cmd = _find_version_cmd(header)
+    return _hex_triplet(version_cmd[1].sdk)
+
+
+def _hex_triplet(version):
     # Parse SDK version number
-    major = (version_cmd[1].sdk & 0xFF0000) >> 16
-    minor = (version_cmd[1].sdk & 0xFF00) >> 8
-    revision = (version_cmd[1].sdk & 0xFF)
+    major = (version & 0xFF0000) >> 16
+    minor = (version & 0xFF00) >> 8
+    revision = (version & 0xFF)
     return major, minor, revision
 
 
+def macosx_version_min(filename: str) -> tuple:
+    """Get the -macosx-version-min used to compile a macOS binary.
+
+    For fat binaries, the minimum version is selected.
+
+    """
+    return min(
+        _hex_triplet(cmd[1].version)
+        for header in MachO(filename).headers
+        for cmd in header.commands if cmd[0].cmd == LC_VERSION_MIN_MACOSX
```

Comment:
If deployment target is 10.14 or newer, `LC_BUILD_VERSION` is emitted instead of `LC_VERSION_MIN_MACOSX` (and `.version` field is changed into `.minos` in the structure). We probably want to handle that as well.

Reply:
Ahh, that's what that's about. I saw in that `_find_version_cmd()` that it used both but the `LC_BUILD_VERSION` command didn't have a version attribute so I assumed that it was meant for something else. I'll recompile with the target set to 10.4 then work out what it uses instead of the version attribute.

Reply:
It has `minos` field instead of `version`.